src/components/InputPanel.tsx

```diff
@@ -1,102 +1,171 @@
-import { useCallback, useContext, useEffect, useMemo, useState } from 'react';
-import InputContext, { INPUT_TYPES } from '../contexts/InputContext';
-import styles from '../css/RMLMappingEditor.module.scss';
-import CodeEditor from './CodeEditor';
-import { ReactComponent as PlusIcon } from '../images/plus.svg';
-import { ReactComponent as DownArrow } from '../images/down-arrow.svg';
+import { useCallback, useContext, useEffect, useMemo, useState } from "react";
+import InputContext, { INPUT_TYPES } from "../contexts/InputContext";
+import styles from "../css/RMLMappingEditor.module.scss";
+import CodeEditor from "./CodeEditor";
+import { ReactComponent as PlusIcon } from "../images/plus.svg";
+import { ReactComponent as DownArrow } from "../images/down-arrow.svg";
 
 const views = {
-  inputs: 'Input Files',
-  // functions: 'Functions',
-}
+  inputs: "Input Files",
+  functions: "Functions",
+};
+type ViewType = keyof typeof views;
+
+const functionList = [
+  {
+    name: "Capitalize",
+    function: (str: string) => str.toUpperCase(),
+  },
+  {
+    name: "Lower case",
+    function: (str: string) => str.toLowerCase(),
+  },
+  {
+    name: "Concatenate",
+    function: (strList: string[], delimiter: string = "-") =>
+      strList.join(delimiter),
+  },
+];
 
 export interface InputPanelProps {
   addNewInput: () => void;
 }
 
 function InputPanel({ addNewInput }: InputPanelProps) {
-  const [view, setView] = useState(views.inputs);
+  const [view, setView] = useState<ViewType>("inputs");
```

Comment:
I don't love that this is now using a copied string rather than a referenced value (eg. referencing a kew of `views`). Maybe we need to create two maps, one called `views` and one called `viewLabels`:
```js
const views = {
    inputs: 'inputs',
    functions: 'functions',
};
const viewLabels = {
    [views.inputs]: 'Input Files',
    [views.functions]: 'Functions',
};

This was you can initialize state with `views.inputs` and use `viewLabels[view]` for the label.